orchagent/copporch.cpp

```diff
@@ -0,0 +1,479 @@
+                SWSS_LOG_DEBUG("obtained cbs:%d", attr.value.u64);
+            }
+            else if (fvField(*i) == copp_policer_cir_field)          
+            {
+                attr.id = SAI_POLICER_ATTR_CIR;
+                attr.value.u64 = std::stoul(fvValue(*i));
+                policer_attribs.push_back(attr);
+                SWSS_LOG_DEBUG("obtained cir:%d", attr.value.u64);
+            }
+            else if (fvField(*i) == copp_policer_pbs_field)          
+            {
+                attr.id = SAI_POLICER_ATTR_PBS;
+                attr.value.u64 = std::stoul(fvValue(*i));
+                policer_attribs.push_back(attr);
+                SWSS_LOG_DEBUG("obtained pbs:%d", attr.value.u64);
+            }
+            else if (fvField(*i) == copp_policer_pir_field)          
+            {
+                attr.id = SAI_POLICER_ATTR_PIR;
+                attr.value.u64 = std::stoul(fvValue(*i));
+                policer_attribs.push_back(attr);
+                SWSS_LOG_DEBUG("obtained pir:%d", attr.value.u64);
+            }
+            else if (fvField(*i) == copp_policer_action_green_field)
+            {
+                SWSS_LOG_DEBUG("green action:%s", queue_ind.c_str());
+                sai_packet_action_t policer_action = packet_action_map.at(fvValue(*i));
+                attr.id = SAI_POLICER_ATTR_GREEN_PACKET_ACTION;
+                attr.value.s32 = policer_action;
+                policer_attribs.push_back(attr);
+            }
+            else if (fvField(*i) == copp_policer_action_red_field) 
+            {
+                SWSS_LOG_DEBUG("red action:%s", queue_ind.c_str());
+                sai_packet_action_t policer_action = packet_action_map.at(fvValue(*i));
+                attr.id = SAI_POLICER_ATTR_RED_PACKET_ACTION;
+                attr.value.s32 = policer_action;
+                policer_attribs.push_back(attr);
+            }
+            else if (fvField(*i) == copp_policer_action_yellow_field)
+            {
+                SWSS_LOG_DEBUG("yellowaction:%s", queue_ind.c_str());
+                sai_packet_action_t policer_action = packet_action_map.at(fvValue(*i));
+                attr.id = SAI_POLICER_ATTR_YELLOW_PACKET_ACTION;
+                attr.value.s32 = policer_action;
+                policer_attribs.push_back(attr);
+            }
+            else
+            {
+                SWSS_LOG_ERROR("Unknown copp field specified:%s\n", fvField(*i).c_str());
+                return task_process_status::task_invalid_entry;
+            }
+        }
+
+        if (m_trap_map.find(key) != m_trap_map.end())
+        {
+            SWSS_LOG_DEBUG("found existing trap_group object:%s", key.c_str());
+            if(!policer_attribs.empty())
+            {
+                attr.id = SAI_HOSTIF_TRAP_GROUP_ATTR_POLICER;
+                attr.value.oid = SAI_NULL_OBJECT_ID;
+                if (SAI_STATUS_SUCCESS != (sai_status = sai_hostif_api->get_trap_group_attribute(m_trap_map[key], 1, &attr)))
```

Comment:
It is not necessary to use get api to retrieve the trap group ID from the SAI as you know if you have created the trap group or not, and should cache the trap group ID. In the orchagent, we should not use get API unless we have to. For example, the hw_lane ID, number of ports, and etc. on. Everything else should be cached in the orchagent.